test/common/http/header_map_impl_test.cc

```diff
@@ -907,6 +910,24 @@ TEST(HeaderMapImplTest, Get) {
   }
 }
 
+TEST(HeaderMapImplTest, TestHeaderList) {
+  std::array<Http::LowerCaseString, 2> keys{Headers::get().Path, LowerCaseString("hello")};
+  std::array<std::string, 2> values{"/", "world"};
+
+  auto headers = createHeaderMap<TestHeaderMapImpl>({{keys[0], values[0]}, {keys[1], values[1]}});
+  const auto header_list = HeaderListView(headers->header_map_);
+  std::vector<absl::string_view> str_header_keys(2);
+  std::transform(header_list.keys().begin(), header_list.keys().end(), str_header_keys.begin(),
+                 [](auto key) -> absl::string_view { return key.get().getStringView(); });
+  std::vector<absl::string_view> str_header_values(2);
+  std::transform(header_list.values().begin(), header_list.values().end(),
+                 str_header_values.begin(),
+                 [](auto value) -> absl::string_view { return value.get().getStringView(); });
+
+  EXPECT_THAT(str_header_keys, ElementsAre(":path", "hello"));
+  EXPECT_THAT(str_header_values, ElementsAre("/", "world"));
```

Comment:
This seems like a lot of transform complexity obscuring the meaning of the test. Maybe something like this would at least reduce the bulk and make it easier to read?
```
auto to_string_views = [](const HeaderListView::HeaderStrings& strs) : std::vector<absl::string_view> {
  std::vector<absl::string_view> str_views;
  std::transform(strs.begin(), strs.end(), str_views.begin(), 
                 [](auto value) -> absl::string_view { return value.get().getStringView(); });
  return str_views;
};

EXPECT_THAT(to_string_views(header_list.keys()), ElementsAre(":path", "hello"));
EXPECT_THAT(to_string_views(header_list.values()), ElementsAre("/", "world"));
```

I think this factoring out of the transform logic makes it easier to read; WDYT?

Reply:
ah, that is a great idea. thanks.